com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -2065,6 +2065,21 @@ internal void HandleConnectionApproval(ulong ownerClientId, ConnectionApprovalRe
 
                     SendMessage(ref message, NetworkDelivery.ReliableFragmentedSequenced, ownerClientId);
 
+                    for (int index = 0; index < MessagingSystem.MessageHandlers.Length; index++)
+                    {
+                        if (MessagingSystem.ReverseTypeMap[index] != null)
+                        {
+                            var orderingMessage = new OrderingMessage
+                            {
+                                Order = index,
+                                Hash = MessagingSystem.ReverseTypeMap[index].FullName.GetHashCode()
```

Comment:
does `.FullName.GetHashCode()` call resolve to [`String.GetHashCode()`](https://docs.microsoft.com/en-us/dotnet/api/system.string.gethashcode)?

if so, we should not use it and pick our `xxHash32` implementation we have inside the package (`/Runtime/Hashing/XXHash/XXHash.cs`).

`xxHash32` hashing would give us fast and stable (but still non-cryptographic) hashes across platforms, versions etc but that would not necessarily be the case with `String.GetHashCode()` which would cause a huge pain for users & us.

Reply:
Can you elaborate on "huge pain for users & us."? On the performance side, it is a one-off at startup. Is this a portability concern as different platform/builds could have different hashes for the same string?

Reply:
mainly debugging, secondly portability.

Reply:
> ...
> The hash code itself is not guaranteed to be stable. Hash codes for identical strings can differ across .NET implementations, across .NET versions, and across .NET platforms (such as 32-bit and 64-bit) for a single version of .NET. In some cases, they can even differ by application domain. This implies that two subsequent runs of the same program may return different hash codes.
>
> As a result, hash codes should never be used outside of the application domain in which they were created, they should never be used as key fields in a collection, and they should never be persisted.
> ...